src/test/scala/ai/privado/languageEngine/csharp/CSharpTestBase.scala

```diff
@@ -0,0 +1,195 @@
+package ai.privado.languageEngine.csharp
+
+import ai.privado.cache.{AuditCache, DataFlowCache, RuleCache, TaggerCache}
+import ai.privado.entrypoint.PrivadoInput
+import ai.privado.model.*
+import ai.privado.threatEngine.ThreatEngineExecutor
+import better.files.File
+import io.joern.csharpsrc2cpg.{CSharpSrc2Cpg, Config}
+import io.joern.dataflowengineoss.language.Path
+import io.shiftleft.codepropertygraph.generated.Cpg
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+import scala.collection.mutable
+import ai.privado.cache.*
+import io.joern.x2cpg.X2Cpg
+import io.shiftleft.semanticcpg.layers.*
+import io.joern.dataflowengineoss.layers.dataflows.*
+import ai.privado.languageEngine.csharp.tagger.source.IdentifierTagger
+
+abstract class CSharpTestBase extends AnyWordSpec with Matchers with BeforeAndAfterAll with BeforeAndAfterEach {
+
+  private val cpgs        = mutable.ArrayBuffer.empty[Cpg]
+  private val outPutFiles = mutable.ArrayBuffer.empty[File]
+  private val inputDirs   = mutable.ArrayBuffer.empty[File]
+
+  val sourceRule: List[RuleInfo] = List(
```

Comment:
@karan-batavia Instead of defining these rules for each test case and Language, we can keep a single copy of the rules at - https://github.com/Privado-Inc/privado-core/blob/main/src/test/scala/ai/privado/RuleInfoTestData.scala